<launch>
  <arg name="robot_ip" />

  <include file="$(find excel_hw_iface)/launch/load_excel_ctrl.launch" />

  <node name="excel_load_ctrls" type="spawner" pkg="controller_manager"
        output="screen" args="--stopped vel_trajectory_ctrl pva_enable_mode" />

  <node name="excel_spawn_ctrls" type="spawner" pkg="controller_manager"
        output="screen" args="joint_state_pub mode_state_pub ur_diagnostics_pub config_fwd_ctrl vel_enable_mode vel_forward_ctrl" />

  <node pkg="diagnostic_aggregator" type="aggregator_node"
      name="diag_agg" >
      <rosparam> 
          pub_rate: 1.0
          analyzers:
              arm:
                  type: AnalyzerGroup
                  path: UR Arm
                  analyzers:
                      controller:
                          type: GenericAnalyzer
                          path: Controller
                          contains: UR arm controller
                      joints:
                          type: GenericAnalyzer
                          path: Joints
                          contains: UR arm joint
      </rosparam> 
  </node>

  <node pkg="excel_hw_iface" type="excel_ctrl_man_xeno" name="excel_ctrl_man_xeno">
    <rosparam command="load" file="$(find excel_hw_iface)/launch/joint_names.yaml"
              param="joint_names" />
    <param name="robot_ip" value="$(arg robot_ip)" />
  </node>

</launch>
